### PR TITLE
Update Microsoft/go-winio v0.4.14

### PR DESCRIPTION
**Pull request by jterry75**
Signed-off-by: Justin Terry (VM) <juterry@microsoft.com>

This revendor is the latest release of Microsoft/go-winio and it fixes a regression in the GraphDriver vhd.DetachVhd.

### CODE REVIEW

**Comment by jterry75** [comment]
@Random-Liu - FYI

**Comment by jterry75** [comment]
@thaJeztah - Powerpc is unrelated.
@jhowardmsft - FYI

**Comment by thaJeztah** [comment]
is this related to https://github.com/docker/for-win/issues/3884,  https://github.com/moby/moby/issues/27588#issuecomment-499484206, and https://github.com/microsoft/hcsshim/issues/624 ?

If so, does this fix those issues, or are platform changes needed as well?

**Comment by jterry75** [comment]
@thaJeztah - Unrelated to all. This is to fix a regression here: https://github.com/moby/moby/blob/master/daemon/graphdriver/windows/windows.go#L343 if the fallback code needs to call `vhd.DetachVhd` due to in-use the call with fail with `E_INVALID_PARAMETER` due to the vendored update to go-winio previously.

**Review by AkihiroSuda — APPROVED** [review]

File: vendor.conf [hunk 1]
```diff
@@ -1,6 +1,6 @@
 github.com/Azure/go-ansiterm                        d6e3b3328b783f23731bc4d058875b0371ff8109
 github.com/Microsoft/hcsshim                        672e52e9209d1e53718c1b6a7d68cc9272654ab5
-github.com/Microsoft/go-winio                       3fe4fa31662f6ede2353d913e93907b8e096e0b6
+github.com/Microsoft/go-winio                       v0.4.14
```
Comment (thaJeztah) [inline vendor.conf]: Could you pin this by digest, and add the tag as a comment after that?

```suggestion
github.com/Microsoft/go-winio                       6c72808b55902eae4c5943626030429ff20f3b63 # v0.4.14
```

**Review by thaJeztah — CHANGES_REQUESTED** [review]
one nit, but looks good otherwise

File: vendor.conf (hunk 1 above)
Comment (jterry75) [reply vendor.conf]: Sure no problem.

**Review by thaJeztah — APPROVED** [review]
LGTM, thanks!

**Comment by thaJeztah** [comment]
ignoring the remaining failure, which is a known flaky test;
https://ci.docker.com/public/blue/organizations/jenkins/moby/detail/PR-39679/2/pipeline

```
[2019-08-07T17:44:16.488Z] FAIL: docker_cli_swarm_test.go:1492: DockerSwarmSuite.TestSwarmNetworkIPAMOptions
[2019-08-07T17:44:16.488Z] 
[2019-08-07T17:44:16.488Z] Creating a new daemon
[2019-08-07T17:44:16.488Z] [d10c12796639e] waiting for daemon to start
[2019-08-07T17:44:16.488Z] [d10c12796639e] waiting for daemon to start
[2019-08-07T17:44:16.488Z] [d10c12796639e] daemon started
[2019-08-07T17:44:16.488Z] 
[2019-08-07T17:44:16.488Z] docker_cli_swarm_test.go:1501:
[2019-08-07T17:44:16.488Z]     c.Assert(strings.TrimSpace(out), checker.Contains, "foo:bar")
[2019-08-07T17:44:16.488Z] ... obtained string = "map[]"
[2019-08-07T17:44:16.488Z] ... substring string = "foo:bar"
[2019-08-07T17:44:16.488Z] 
[2019-08-07T17:44:16.488Z] [d10c12796639e] Stopping daemon
[2019-08-07T17:44:16.488Z] [d10c12796639e] exiting daemon
[2019-08-07T17:44:16.488Z] [d10c12796639e] Daemon stopped
```

**Comment by thaJeztah** [comment]
@jterry75 the regression was in https://github.com/moby/moby/pull/38922 (or in https://github.com/moby/moby/pull/39124 ?) I guess the first one, as it modifies the vhd code

**Comment by thaJeztah** [comment]
Opened a backport for 19.03 https://github.com/docker/engine/pull/314

File: vendor/github.com/Microsoft/go-winio/go.mod [hunk 2]
```diff
@@ -0,0 +1,9 @@
+module github.com/Microsoft/go-winio
+
+go 1.12
+
+require (
+	github.com/pkg/errors v0.8.1
+	github.com/sirupsen/logrus v1.4.1
+	golang.org/x/sys v0.0.0-20190507160741-ecd444e8653b
```
Comment (thaJeztah) [inline vendor/github.com/Microsoft/go-winio/go.mod]: fwiw; double checked if this needed updating, but we're already ahead of this commit in master; https://github.com/golang/sys/compare/ecd444e8653b...4c4f7f33c9ed00de01c4c741d2177abfcfe19307

**Pull request merged**